### PR TITLE
pg-pool: Implemented new types for the pg-pool package

### PR DESCRIPTION
**Pull request by mainnika**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/brianc/node-pg-pool/blob/master/index.js
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by aleung — APPROVED** [review]

File: types/pg-pool/tslint.json [hunk 1]
```diff
@@ -1,80 +1,6 @@
...
-        "space-before-function-paren": false,
-        "space-within-parens": false,
-        "strict-export-declare-modifiers": false,
-        "trim-file": false,
-        "triple-equals": false,
-        "typedef-whitespace": false,
-        "unified-signatures": false,
-        "void-return": false,
-        "whitespace": false
+        "strict-export-declare-modifiers": false
```
Comment (sheetalkamat) [inline types/pg-pool/tslint.json]: Remove this rule and fix errors

File: types/pg-pool/index.d.ts [hunk 2]
```diff
@@ -1,7 +1,28 @@
...
 // Project: https://github.com/brianc/node-pg-pool
-// Definitions by: Leo Liang <https://github.com/aleung>
+// Definitions by: Leo Liang <https://github.com/aleung>, Nikita Tokarchuk <https://github.com/mainnika>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-/// <reference types="pg" />
-export {Pool, PoolConfig } from "pg";
+import * as pg from 'pg';
+
+declare class Pool<T extends pg.Client> extends pg.Pool {
```
Comment (sheetalkamat) [inline types/pg-pool/index.d.ts]: Extend EventEmitter here instead.

File: types/pg-pool/index.d.ts [hunk 3]
```diff
@@ -1,7 +1,28 @@
...
+    readonly Client: Pool.ClientLikeCtr<T>;
+
+    constructor(config?: Pool.Config<T>, client?: Pool.ClientLikeCtr<T>);
+
+    connect(): Promise<T & pg.PoolClient>;
+    connect(callback: (err: Error, client: T & pg.PoolClient, done: (release?: any) => void) => void): void;
```
Comment (sheetalkamat) [inline types/pg-pool/index.d.ts]: Needs `query` ? https://github.com/brianc/node-pg-pool#your-new-favorite-helper-method

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/pg-pool/index.d.ts (hunk 2 above)
Comment (mainnika) [reply types/pg-pool/index.d.ts]: I can either move the entire pg.Pool types here or keep it like this.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/pg/index.d.ts#L142

File: types/pg-pool/index.d.ts (hunk 3 above)
Comment (mainnika) [reply types/pg-pool/index.d.ts]: Implemented in the `pg` types, https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/pg/index.d.ts#L158

File: types/pg-pool/tslint.json (hunk 1 above)
Comment (mainnika) [reply types/pg-pool/tslint.json]: removed 1641db0ed4649e1f01eeb227d417cad986995800

File: types/pg-pool/index.d.ts (hunk 2 above)
Comment (sheetalkamat) [reply types/pg-pool/index.d.ts]: Sorry about that I didn't notice that pg.Pool does extend EventEmitter

**Pull request merged**